cpp/src/arrow/util/rle_encoding_test.cc

```diff
@@ -602,20 +1033,106 @@ struct GetBatchSpacedTestCase {
+      {
+          {
+              RandomPart{/* max=*/7, /* size=*/10037, /* null_proba= */ 0.1},
+              NullPart{/* size= */ 1153},
+              RandomPart{/* max=*/7, /* size=*/800, /* null_proba= */ 0.5},
+          },
+          /* bit_width= */ 3,
+      },
+      {
+          {
+              NullPart{/* size= */ 80},
+              RandomPart{/* max=*/1023, /* size=*/800, /* null_proba= */ 0.01},
+              NullPart{/* size= */ 1023},
+          },
+          /* bit_width= */ 11,
+      },
+      {
+          {RepeatPart{/* value=*/13, /* size=*/100000}},
+          /* bit_width= */ 10,
+      },
+      {
+          {
+              NullPart{/* size= */ 1024},
+              RepeatPart{/* value=*/10000, /* size=*/100000},
+              NullPart{/* size= */ 77},
+          },
+          /* bit_width= */ 23,
+      },
+      {
+          {
+              RepeatPart{/* value=*/13, /* size=*/100000},
+              NullPart{/* size= */ 1153},
+              RepeatPart{/* value=*/72, /* size=*/100799},
+          },
+          /* bit_width= */ 10,
+      },
+      {
+          {
+              RandomPart{/* max=*/1, /* size=*/1013, /* null_proba= */ 0.01},
+              NullPart{/* size=*/8},
+              RepeatPart{1, /* size= */ 256},
+              NullPart{/* size=*/128},
+              RepeatPart{0, /* size= */ 256},
+              NullPart{/* size=*/15},
+              RandomPart{/* max=*/1, /* size=*/8 * 1024, /* null_proba= */ 0.01},
+          },
+          /* bit_width= */ 1,
+      },
   };
-  for (auto case_ : int32_cases) {
-    auto arr = rand.Int32(case_.size, /*min=*/0, case_.max_value, case_.null_probability);
-    CheckRoundTripSpaced<Int32Type>(*arr, case_.bit_width);
-    CheckRoundTripSpaced<Int32Type>(*arr->Slice(1), case_.bit_width);
+
+  ::arrow::random::RandomArrayGenerator rand(/* seed= */ 12);
+  // FRAGILE: Large enough so that it can be indexed by any value in all cases
+  auto dict = std::static_pointer_cast<arrow::FloatArray>(rand.Float32(20000, -1.0, 1.0));
+
+  for (auto case_ : test_cases) {
+    if (static_cast<std::size_t>(case_.bit_width) > sizeof(T)) {
+      continue;
+    }
+
+    auto array = case_.MakeArray(rand);
+
+    // Tests for GetBatch
+    CheckRoundTrip<ArrowType, false, 1>(*array, case_.bit_width);
+    CheckRoundTrip<ArrowType, false, 3>(*array, case_.bit_width);
+
+    // Tests for GetBatchSpaced
+    CheckRoundTrip<ArrowType, true, 1>(*array, case_.bit_width);
+    CheckRoundTrip<ArrowType, true, 7>(*array, case_.bit_width);
+    CheckRoundTrip<ArrowType, true, 1>(*array->Slice(1), case_.bit_width);
+
+    // Cannot test GetBatchWithDict with this method since unknown null values
+
+    // Tests for GetBatchWithDictSpaced
+    CheckRoundTrip<ArrowType, true, 1>(*array, case_.bit_width, dict);
+    CheckRoundTrip<ArrowType, true, 5>(*array, case_.bit_width, dict);
   }
 }
 
+TEST(RleBitPacked, GetBatchSpacedRoundtripUint16) {
+  DoTestGetBatchSpacedRoundtrip<uint16_t>();
```

Comment:
We should also test `uint8_t`. Also, do we want to test signed integers or is that not supported by the RleBitPackedDecoder?